### ISSUE TITLE
Feature request : Replace fanart with video trailer and skip intro

### ISSUE THREAD

**Issue by Dodobibi** [asciidisco/plugin.video.netflix#382]
*I'm submitting a ...*
  - [ ] bug report
  - [x] feature request
  - [ ] support request

It will be great if (like in the native app) :
- addon can replace Netflix fanart with video trailer
- Show a button to skip introduction when play

I can help for coding this, but don't know where to begin.

**Comment by liberty-developer** [comment]
From what I know this would require a c++ View Implementation.
Nothing impossible, but most likely some work.

@Dodobibi how are your c++ skills?

**Comment by Dodobibi** [comment]
@liberty-developer  My c++ skills : medium (It's not my favorite language)

However, tvtunes already do the job without c++ (for video fanart, it play a video in background).
I'm beginning to do the same way (playing video in background). I search howto construct the url of netflix "preplay" to pass to kodi.

For trailer (or preplay) :
- First : modify or create a new call to netflix api for retrieve the video id of the trailer _(I don't know howto)_
- Second : build the url of the trailer _(I will find howto)_
- Thirdly : implement play / stop trailer when navigating _(I know howto, no need to c++ view for that. **Am I wrong ?**)_
@liberty-developer Why do you think this would require a c++ view ?

For skip intro :
- First : modify or create a new call to netflix api for retrieve the begin time and end time of the intro_(I don't know howto)_
- Second : show a banner with "Skip introduction ? Yes / No" when playing in this interval _(I know howto)_
- Thirdly : jump to end of intro if user respond yes _(I know howto)_

I create a single feature request for both of these, because my problem is identique : howto do call on netlflix api to retrieve additional information about the video ?

**Comment by im85288** [comment]
Hi guys, dev of the next up addon here. We started a while back thinking of integrating this addon with mine but did not get to the stage where it was completed. I’d like to finish this off if this is now complete. Just let me know. Cheers

**Comment by Dodobibi** [comment]
@im85288 I made a PR #395 
For skipping intro / recap, I not use your addon because it's just a jump in the video.
However, I can use it for jump to next episode at the end, but I not find how to pass informations to it. Can you explain me how to use it please ?

**Comment by im85288** [comment]
@Dodobidi - the work was done by @hawkeyexp where he set up some window properties that need populated with the start time and end time of the notification. I would need to have a look at the code again to get familiar as it’s been some time but it goes along with what @caphm started about populating Window properties with the times.

**Comment by Dodobibi** [comment]
@im85288 I not find any windows properties to set times in next up addon source code. Or i'm not on the good repository ?
Where can I find their work (@hawkeyexp & @caphm)?

**Comment by im85288** [comment]
@Dodobibi - the changes are currently on the development branch but some further work is needed there and at the time it was only using a local skipintro db and not this addon for the times. The last I heard from @hawkeyexp was here: https://github.com/im85288/service.nextup.notification/issues/55 I believe he is a dev on this add-on so hopefully will not be long before he reads this.

It was pretty simple in it's implementation as can be seen here: https://github.com/im85288/service.nextup.notification/blob/development/resources/lib/skip_intro.py in the UI part. This window pops up the skip into button and on press seeks to the times as defined in the window properties found here: https://github.com/im85288/service.nextup.notification/blob/development/resources/lib/common.py specifically the NextUpNotification.intro_start and NextUpNotification.intro_length

I think we should do this properly though and not use window properties but instead take advantage of addon communication using the Addon Signals api: https://github.com/ruuk/script.module.addon.signals I started using that for integration with an emby addon where we did not want to depend on each other. 

If you were to implement something similar for this addon...where it sends signals for the intro time, length etc then any other addon that is interested (in my case the next up one) can listen for those changes and re-act accordingly.

So what I propose is something along these lines:

`#In source addon
import AddonSignals
​
AddonSignals.sendSignal('NETFLIX_INTRO_DATA', {'start': '1min 40 sec','end':'2min 30 sec','length':'50 sec'})`


So basically your firing off a signal with a bunch of JSON data describing the episode into start time etc

The I or any other addon can listen for that signal and do what is needed.

Does that make sense? It's something I could do if you like too or if you have a better idea happy to hear it.
